online/src/main/scala/ai/chronon/online/FetcherBase.scala

```diff
@@ -511,7 +521,7 @@ class FetcherBase(kvStore: KVStore,
                 joinConf: Option[Join] = None): Future[scala.collection.Seq[Response]] = {
     val startTimeMs = System.currentTimeMillis()
     // convert join requests to groupBy requests
-    val joinDecomposed: scala.collection.Seq[(Request, Try[Seq[Either[PrefixedRequest, KeyMissingException]]])] =
+    val joinDecomposed: scala.collection.Seq[(Request, Try[Seq[Either[PrefixedRequest, FetchException]]])] =
       requests.map { request =>
         val joinTry: Try[JoinOps] = if (joinConf.isEmpty) {
```

Comment:
i think we should do the check at this line, before we are making the `getJoinConf` call. the `getJoinConf` may still get stale data for inactive join.

Reply:
Hi @hzding621 We need to get joinConf first to get the owningTeam to validate the fetched join. I think it is fine to get stale data here, if the join is invalid the real data fetch won't happen then.

Reply:
Makes sense. Do you mind leave a comment?